.github/workflows/check.yml

```diff
@@ -155,3 +157,32 @@ jobs:
         run: rustup component add rust-src
       - name: Run tests with sanitizer
         run: RUSTFLAGS="-Z sanitizer=${{ matrix.sanitizer }} -Z sanitizer-memory-track-origins" cargo test -Z build-std --target $TARGET --no-default-features --features "cli web-app real-world-infra test-fixture descriptive-gate"
+
+  coverage:
+    name: Measure coverage
+    runs-on: ubuntu-latest
+    env:
+      CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
You should be able to pass this directly to the action rather than putting it in the environment.  See below.